### DISCUSSION TITLE
Add leader-line js library to NiceGUI

### DISCUSSION THREAD

**Discussion by yanghw116** [zauberzeug/nicegui#2443] [Ideas / Feature Requests]
I'm new to nicegui. I found a very nice js lib named [leader-line](https://github.com/anseki/leader-line). Unfortunately, I don't use js (that's why I tend to nicegui). Could anyone help me to pack it so that it can be used in nicegui? Thanks in advance.

**Comment by falkoschindler** [comment]
Hi @yanghw116,

This looks really nice! But do you know why [the repository](https://github.com/anseki/leader-line) is archived? The last commit is from two years ago. Maybe there is a more active fork?

Anyway, it would be great to have a NiceGUI integration. Here's a proof of concept:

```py
ui.add_head_html('<script src="https://cdnjs.cloudflare.com/ajax/libs/leader-line/1.0.7/leader-line.min.js"></script>')

with ui.row().classes('w-full justify-between'):
    with ui.card() as a:
        ui.label('A')
    with ui.card() as b:
        ui.label('B')

ui.add_body_html(f'''
    <script>
        document.addEventListener('DOMContentLoaded', function() {{
            const line = new LeaderLine(
                getElement({a.id}).$el,
                getElement({b.id}).$el,
            );
        }});
    </script>
''')
```

<img width="490" alt="Screenshot 2024-01-26 at 07 43 53" src="https://github.com/zauberzeug/nicegui/assets/5767091/1020bec4-0683-49b5-87fc-f90b7ac345c6">

**Reply by yanghw116** [reply]
I have no idea why it is closed. 

About the example code, here raised another question. how do I get the object of the arrowline (like `a` and `b` for the boxes), so that I can handle it in python scripts? for example, I want to change the color of the arrowline when I hit a button.
